src/components/implementation/no_interface/scheddev/sl_lock.h

```diff
@@ -0,0 +1,23 @@
+#ifndef SL_LOCK_H
+#define SL_LOCK_H
+
+#include <cos_kernel_api.h>
+
+struct sl_lock {
+    // will be 0 if no one holds the lock
+    volatile thdid_t holder;
+};
+
+#define SL_LOCK_STATIC_INIT() (struct sl_lock) { .holder = 0 }
+
+void sl_lock_init(struct sl_lock *lock);
+
+thdid_t sl_lock_holder(struct sl_lock *lock);
+
+void sl_lock_lock(struct sl_lock *lock);
```

Comment:
These are common operations, and I think that we want to give the compiler the opportunity to optimize them.  I think `sl_lock_take` and `sl_lock_release` should be `static inline` in the header.  I see little to no downside to doing this, and it enables inlining.